### PR TITLE
[BUILD] Remove option WITH_OTLP

### LINKED ISSUE #1926: [BUILD] CMake option WITH_OTLP_GRPC not predictable, sometime not displayed.
As seen with PR #1916, CMake option `WITH_OTLP_GRPC` needs some cleanup.

It should not be a `cmake_dependent_option()`, just a plain` option()`.

### PR DESCRIPTION
**Pull request by ThomsonTan**
Fixes #1926, #2160.

## Changes

Remove `WITH_OTLP` option and raise error on its definition as it causes confusions (see the above issues). Also make `WITH_OTLP_GRPC` as a plain option like `WITH_OTLP_HTTP`.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -159,12 +159,10 @@ if(WITH_STL)
   endif()
 endif()
 
-option(WITH_OTLP "Whether to include the OpenTelemetry Protocol in the SDK" OFF)
-
-if(WITH_OTLP)
+if(DEFINED WITH_OTLP)
```
Comment (owent) [inline CMakeLists.txt:162]: Could we add  `option(WITH_OTLP_GRPC "Whether to include the OTLP gRPC exporter in the SDK" OFF)` here? It's related to #1926 .

File: ci/do_ci.sh [hunk 2]
```diff
@@ -141,7 +141,6 @@ elif [[ "$1" == "cmake.maintainer.cpp11.async.test" ]]; then
   rm -rf *
   cmake -DCMAKE_BUILD_TYPE=Debug  \
         -DCMAKE_CXX_STANDARD=11 \
-        -DWITH_OTLP=ON \
```
Comment (owent) [inline ci/do_ci.sh:146]: Should we use `-DWITH_OTLP_GRPC=ON` here?

**Review by owent — APPROVED** [review]
LGTM and thanks.

**Review by lalitb — APPROVED** [review]

File: INSTALL.md [hunk 3]
```diff
@@ -82,7 +82,7 @@ You can link OpenTelemetry C++ SDK with libraries provided in
    - `-DBUILD_SHARED_LIBS=ON` : To build shared libraries for the targets.
       Please refer to note [below](#building-shared-libs-for-windows) for
       Windows DLL support.
-   - `-DWITH_OTLP=ON` : To enable building OTLP exporter.
+   - `-DWITH_OTLP_GRPC=ON` : To enable building OTLP exporter.
```
[the remark this answers is not in the document]
Comment (ThomsonTan) [reply INSTALL.md]: Thanks for the catch.

**Pull request merged**